src/thread/id.rs

```diff
@@ -12,6 +11,61 @@ use crate::process::Pid;
 /// [Linux]: https://man7.org/linux/man-pages/man2/gettid.2.html
 #[inline]
 #[must_use]
-pub fn gettid() -> Pid {
+#[cfg(any(target_os = "android", target_os = "linux"))]
+pub fn gettid() -> crate::process::Pid {
     backend::thread::syscalls::gettid()
 }
+
+/// `setuid(uid)`—Sets the effective user ID of the calling process.
+///
+/// # Warning
+///
+/// On Linux, this only changes the xid of the current *thread*,
+/// not the entire process even though that is in violation of the POSIX standard.
+///
+/// # References
+///  - [POSIX]
+///  - [Linux]
+///
+/// [POSIX]: https://pubs.opengroup.org/onlinepubs/9699919799/functions/setuid.html
+/// [Linux]: https://man7.org/linux/man-pages/man2/setuid.2.html
+#[inline]
+#[must_use]
+#[cfg(not(any(target_os = "redox", target_os = "wasi")))]
+pub fn setuid(uid: crate::process::Uid) -> io::Result<()> {
```

Comment:
I propose renaming this function to `set_thread_uid`, having it only be defined on `cfg(any(target_os = "android", target_os = "linux"))`, and rephrasing the warning from "On Linux, [it's different]" to "This is not `setuid`; here's now it's different".

Reply:
And similar for `setgid`.

Reply:
So we leave other platforms out?

Reply:
If any other platform has a way to set a thread-specific uid, then we can implement it for them. I don't know of any others that do.

Does your use case need POSIX-semantics `setuid`? If so, then we add a `rustix::process::setuid` which always calls libc. (Or, we implement the POSIX semantics manually for linux_raw, but I'm assuming that's not worthwhile at this point.)

Reply:
Gotya, so we're going with the "linux is weird and gets special treatment, we'll deal with other platforms later" approach. Seems reasonable to me. And yeah, I only need linux support.

Reply:
The ideal is that any given name has at most one meaning. Rustix has tons of functions that are available on some OS's and not others. But we want to avoid the same name meaning different things on different platforms.

That said, there are gray areas. There are platform-specific variations in lots of mostly-portable functions. So we have to make a judgement call whether something is "different enough" to justify giving it a different name. In this case, setting the uid for just one thread instead of all threads feels definitely "different enough".

Reply:
Gotya, thanks for the explanation. :+1: